src/parachain/transaction.ts

```diff
@@ -49,12 +70,36 @@ export class DefaultTransactionAPI implements TransactionAPI {
     async sendLogged<T extends AnyTuple>(
         transaction: SubmittableExtrinsic<"promise">,
         successEventType?: AugmentedEvent<ApiTypes, T>,
-        onlyInBlock?: boolean
+        extrinsicStatus?: ExtrinsicStatus
     ): Promise<ISubmittableResult> {
         if (this.account === undefined) {
             return Promise.reject(new Error(ACCOUNT_NOT_SET_ERROR_MESSAGE));
         }
-        return DefaultTransactionAPI.sendLogged(this.api, this.account, transaction, successEventType, onlyInBlock);
+        return DefaultTransactionAPI.sendLogged(this.api, this.account, transaction, successEventType, extrinsicStatus);
+    }
+
+    async getFeeEstimation(extrinsic: SubmittableExtrinsic<"promise">): Promise<MonetaryAmount<Currency>> {
+        const nativeCurrency = tokenSymbolToCurrency(this.api.consts.currency.getNativeCurrencyId.asToken);
+
+        const account = this.account;
+        if (account === undefined) {
+            return newMonetaryAmount(0, nativeCurrency);
+        }
+
+        const paymentInfo = await extrinsic.paymentInfo(account);
+        return newMonetaryAmount(paymentInfo.partialFee.toString(), nativeCurrency);
+    }
+
+    async dryRun(extrinsic: SubmittableExtrinsic<"promise">): Promise<boolean> {
+        if (this.account === undefined) {
+            return Promise.reject(new Error(ACCOUNT_NOT_SET_ERROR_MESSAGE));
+        }
+        try {
+            await extrinsic.dryRun(this.account);
+            return true;
+        } catch {
+            return false;
+        }
```

Comment:
Is there a way to capture / get a preview of the error message?

If so, I think it might make sense to return that in some way.

Reply:
Good idea, I changed the return type to `{ success: bool, error?: unknown}` to be able to handle the error later (display in UI, log or so).